### PR TITLE
Add custom fluent assertions for Maybe object

### PR DESCRIPTION
**Pull request by tomrijnbeek**

### CODE REVIEW

File: Bearded.Utilities.Testing/Core/MaybeAssertions.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public sealed class MaybeAssertions<T>
+    {
+        private readonly Maybe<T> subject;
+
+        public MaybeAssertions(Maybe<T> instance) => subject = instance;
+        
+        [CustomAssertion]
+        public void BeJust(T value, string because = "", params object[] becauseArgs)
+        {
+            BeJust().Which.Should().BeEquivalentTo(value, because, becauseArgs);
```
Comment (amulware) [inline Bearded.Utilities.Testing/Core/MaybeAssertions.cs]: Why equivalent? I'd expect this to test equality.

File: Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            var maybe = Maybe.Just(100);
+
+            Action assertion = () => maybe.Should().BeJust().Which.Should().Be(200);
+            
+            assertion.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void BeJust_SucceedsWhenComparingToMaybeWithRightValue()
```
Comment (amulware) [inline Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs]: It's comparing to the right value, not to a maybe with the right value, no?

File: Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            var maybe = Maybe.Just(100);
+
+            Action assertion = () => maybe.Should().BeJust(100);
+            
+            assertion.Should().NotThrow();
+        }
+
+        [Fact]
+        public void BeJust_FailsWhenComparingToMaybeWithWrongValue()
```
Comment (amulware) [inline Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs]: similar here

File: Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            var maybe = Maybe.Just(100);
+
+            Action assertion = () => maybe.Should().BeJust(200);
+            
+            assertion.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void BeJust_FailsWhenComparingToNothing()
```
Comment (amulware) [inline Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs]: Ok, now I see what you're doing with the names, but this seems inverted to me. What's the reasoning?

File: Bearded.Utilities.Testing/Core/MaybeAssertions.cs (hunk 1 above)
Comment (tomrijnbeek) [reply Bearded.Utilities.Testing/Core/MaybeAssertions.cs]: I thought this was recommended practice, but looking at the documentation again, it seems like they actually discourage it. Changed.

File: Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs (hunk 2 above)
Comment (tomrijnbeek) [reply Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs]: It took me a while to understand this comment, but yeah, you're right. Updated all the test names to hopefully be clearer.

File: Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs (hunk 3 above)
Comment (tomrijnbeek) [reply Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs]: idem

File: Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs (hunk 4 above)
Comment (tomrijnbeek) [reply Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs]: idem

**Pull request merged**